Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java

```diff
@@ -349,39 +349,51 @@ public void unassignTaskFromUser(Task task) throws DataException {
      *
      * @param comment as Comment object
      */
-    public void reportProblem(Comment comment) throws DataException {
+    public void reportProblem(Comment comment, TaskEditType taskEditType) throws DataException {
         Task currentTask = comment.getCurrentTask();
         if (currentTask.isTypeImagesRead() || currentTask.isTypeImagesWrite()) {
             this.webDav.uploadFromHome(getCurrentUser(), comment.getProcess());
         }
         Date date = new Date();
         currentTask.setProcessingStatus(TaskStatus.LOCKED);
-        currentTask.setEditType(TaskEditType.MANUAL_SINGLE);
+        currentTask.setEditType(taskEditType);
         currentTask.setProcessingTime(date);
         taskService.replaceProcessingUser(currentTask, getCurrentUser());
         currentTask.setProcessingBegin(null);
         taskService.save(currentTask);
 
-        Task correctionTask = comment.getCorrectionTask();
-        correctionTask.setProcessingStatus(TaskStatus.OPEN);
-        correctionTask.setProcessingEnd(null);
-        correctionTask.setCorrection(true);
-        taskService.save(correctionTask);
-
-        lockTasksBetweenCurrentAndCorrectionTask(currentTask, correctionTask);
+        if (comment.hasSeparateCorrectionTask()) {
+            Task correctionTask = comment.getCorrectionTask();
+            correctionTask.setProcessingStatus(TaskStatus.OPEN);
+            correctionTask.setProcessingEnd(null);
+            correctionTask.setCorrection(true);
+            taskService.save(correctionTask);
+            lockTasksBetweenCurrentAndCorrectionTask(currentTask, correctionTask);
+        }
         updateProcessSortHelperStatus(currentTask.getProcess());
     }
 
     /**
      * Unified method for solve problem.
      *
      * @param comment
-     *              as Comment object
+     *         as Comment object
      */
-    public void solveProblem(Comment comment) throws DataException, DAOException, IOException {
-        closeTaskByUser(comment.getCorrectionTask());
+    public void solveProblem(Comment comment, TaskEditType taskEditType)
+            throws DataException, DAOException, IOException {
+        if (comment.hasSeparateCorrectionTask()) {
```

Comment:
See above. Not sure if "correction comments" make sense if their `currentTask` and `correctionTask` are the same.